File: src/main/com/stockcomp/controller/SchedulerController.java
package com.stockcomp.controller;

import com.stockcomp.entity.contest.Contest;
import com.stockcomp.request.CreateContestRequest;
import com.stockcomp.service.ContestService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/scheduler")
public class SchedulerController {

    private final ContestService contestService;

    public SchedulerController(ContestService contestService) {
        this.contestService = contestService;
    }

    @PostMapping("/create-contest")
    public ResponseEntity<Contest> createContest(@RequestBody CreateContestRequest request){
        var contest = contestService.createContest(request);

        return ResponseEntity.ok(contest);
    }

    @PostMapping("/start-contest")
    public ResponseEntity<HttpStatus> startContest(@RequestParam("contestNumber") Integer contestNumber) {
        contestService.startContest(contestNumber);

        return new ResponseEntity<>(HttpStatus.OK);
    }

    @PostMapping("/stop-contest")
    public ResponseEntity<HttpStatus> stopContest(@RequestParam("contestNumber") Integer contestNumber) {
        contestService.stopContest(contestNumber);

        return new ResponseEntity<>(HttpStatus.OK);
    }
}

File: src/main/com/stockcomp/controller/SchedulerController.kt
package com.stockcomp.controller

import com.stockcomp.domain.contest.Contest
import com.stockcomp.request.CreateContestRequest
import com.stockcomp.service.ContestService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/scheduler")
class SchedulerController(private val contestService: ContestService) {

    @PostMapping("/create-contest")
    fun createContest(@RequestBody request: CreateContestRequest): ResponseEntity<Contest> {
        val contest = contestService.createContest(request)

        return ResponseEntity.ok(contest)
    }

    @PostMapping("/start-contest")
    fun startContest(@RequestParam("contestNumber") contestNumber: Int): ResponseEntity<HttpStatus> {
        contestService.startContest(contestNumber)

        return ResponseEntity(HttpStatus.OK)
    }

    @PostMapping("/stop-contest")
    fun stopContest(@RequestParam("contestNumber") contestNumber: Int): ResponseEntity<HttpStatus> {
        contestService.stopContest(contestNumber)

        return ResponseEntity(HttpStatus.OK)
    }
}